features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/DocumentEnricher.java

```diff
@@ -204,19 +209,19 @@ public int hashCode() {
 
     protected static ClassificationRequest createClassificationRequest(FlowDocument document) {
         final ClassificationRequest request = new ClassificationRequest();
-        request.setProtocol(Protocols.getProtocol(document.getProtocol()));
+        request.setProtocol(document.getProtocol() == null ? null : Protocols.getProtocol(document.getProtocol()));
         request.setLocation(document.getLocation());
         request.setExporterAddress(document.getHost());
-        if (document.getDirection() == Direction.INGRESS) {
-            request.setDstAddress(document.getDstAddr());
-            request.setDstPort(document.getDstPort());
-            request.setSrcAddress(document.getSrcAddr());
-            request.setSrcPort(document.getSrcPort());
-        } else {
+        if (document.getDirection() == Direction.EGRESS) {
             request.setSrcAddress(document.getDstAddr());
             request.setSrcPort(document.getDstPort());
             request.setDstAddress(document.getSrcAddr());
             request.setDstPort(document.getSrcPort());
+        } else {
```

Comment:
Can we tackle this in a separate issue? 
It would also be nice to have a test available verifying this.

Reply:
test added